modules/tensor_mechanics/src/materials/ComputeElasticSmearedCrackingStress.C

```diff
@@ -0,0 +1,634 @@
+
+      RealTensorValue Q;
+      Q(0,0) = RT(0,0);
+      Q(0,1) = RT(0,1);
+      Q(0,2) = RT(0,2);
+
+      Q(1,0) = RT(1,0);
+      Q(1,1) = RT(1,1);
+      Q(1,2) = RT(1,2);
+
+      Q(2,0) = RT(2,0);
+      Q(2,1) = RT(2,1);
+      Q(2,2) = RT(2,2);
+
+      // Firstly rotate the elasticity tensor into crack frame coordinate
+      _local_elasticity_tensor.rotate(Q);
+
+      // Then update the elasticity tensor in crack frame
+      const RealVectorValue & c = _crack_flags_local;
+      const Real c0_coupled( c(0) < 1 ? 0 : 1 );
+      const Real c1_coupled( c(1) < 1 ? 0 : 1 );
+      const Real c2_coupled( c(2) < 1 ? 0 : 1 );
+
+      const Real c01( c0_coupled * c1_coupled );
+      const Real c02( c0_coupled * c2_coupled );
+      const Real c12( c1_coupled * c2_coupled );
+      const Real c012( c0_coupled * c12 );
+
+      const Real ym = _youngs_modulus;
+      Real val[3];
+      val[0] = _local_elasticity_tensor(0, 0, 0, 0);
+      val[1] = _local_elasticity_tensor(1, 1, 1, 1);
+      val[2] = _local_elasticity_tensor(2, 2, 2, 2);
+
+      // Assume Poisson's ratio goes to zero for the cracked direction.
+      _local_elasticity_tensor(0, 0, 0, 0) = (c(0) < 1 ? c(0) * ym : val[0]);
+      _local_elasticity_tensor(0, 0, 1, 1) *= c01;
+      _local_elasticity_tensor(0, 0, 2, 2) *= c02;
+      _local_elasticity_tensor(0, 0, 1, 2) *= c01;
+      _local_elasticity_tensor(0, 0, 0, 2) *= c012;
+      _local_elasticity_tensor(0, 0, 0, 1) *= c02;
+
+      _local_elasticity_tensor(1, 1, 1, 1) = (c(1) < 1 ? c(1) * ym : val[1]);
+      _local_elasticity_tensor(1, 1, 2, 2) *= c12;
+      _local_elasticity_tensor(1, 1, 1, 2) *= c01;
+      _local_elasticity_tensor(0, 2, 1, 1) *= c12;
+      _local_elasticity_tensor(0, 1, 1, 1) *= c012;
+
+      _local_elasticity_tensor(2, 2, 2, 2) = (c(2) < 1 ? c(2) * ym : val[2]);
+      _local_elasticity_tensor(1, 2, 2, 2) *= c012;
+      _local_elasticity_tensor(0, 2, 2, 2) *= c12;
+      _local_elasticity_tensor(0, 1, 2, 2) *= c02;
+
+      _local_elasticity_tensor(1, 2, 1, 2) *= c01;
+      _local_elasticity_tensor(0, 2, 1, 2) *= c012;
+      _local_elasticity_tensor(0, 1, 1, 2) *= c012;
+
+      _local_elasticity_tensor(0, 2, 0, 2) *= c12;
+      _local_elasticity_tensor(0, 1, 0, 2) *= c012;
+
+      _local_elasticity_tensor(0, 1, 0, 1) *= c02;
+
+      //fill in from symmetry relations
+      for (unsigned int i = 0; i < 3; ++i)
+        for (unsigned int j = 0; j < 3; ++j)
+          for (unsigned int k = 0; k < 3; ++k)
+            for (unsigned int l = 0; l < 3; ++l)
+              _local_elasticity_tensor(i, j, l, k) =
+              _local_elasticity_tensor(j, i, k, l) =
+              _local_elasticity_tensor(j, i, l, k) =
+              _local_elasticity_tensor(k, l, i, j) =
+              _local_elasticity_tensor(l, k, j, i) =
+              _local_elasticity_tensor(k, l, j, i) =
+              _local_elasticity_tensor(l, k, i, j) = _local_elasticity_tensor(i, j, k, l);
```

Comment:
This duplicates `RankFourTensor::fillSymmetricFromInputVector` with 21 entries if I'm not mistaken